### PR TITLE
Fix sorting on PS4

### PR DESCRIPTION
**Pull request by EvgeniiG**
### Purpose of this PR
Fix PS4 build

Katana: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=bugfix&automation-tools_branch=add-platform-filter&unity_branch=2018.3%2Fstaging

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs [hunk 1]
```diff
@@ -218,7 +217,15 @@ public int UpdateShadowRequest(Camera camera, HDShadowInitParameters initParamet
...
             // Create shadow requests array using the light type
             if (shadowRequests == null || shadowRequests.Length != GetShadowRequestCount(shadowSettings))
-                shadowRequests = Enumerable.Range(0, GetShadowRequestCount(shadowSettings)).Select(i => new HDShadowRequest()).ToArray();
+            {
+                int count      = GetShadowRequestCount(shadowSettings);
+                shadowRequests = new HDShadowRequest[count];
+
+                for (int i = 0; i < count; i++)
+                {
+                    shadowRequests[i] = new HDShadowRequest();
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs]: isn't this producing GC alloc? Shoudln't we preallocate a tab and just resize it to max? (this is a questoin more for @alelievr I suppose)

**Comment by sebastienlagarde** [comment]
Katana link?

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs (hunk 1 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs]: ok, @alelievr you can update the code in a separate PR after we merge this one, thanks.

Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs]: Yes

**Comment by EvgeniiG** [comment]
Added the Katana link.

**Pull request merged**